infer/src/integration/JsonReports.ml

```diff
@@ -140,6 +178,80 @@ end) : Printer with type elt = P.elt = struct
         ()
 end
 
+module type PrinterSarif = sig
+  type elt
+
+  val pp_open : F.formatter -> unit -> unit
+
+  val pp_results : F.formatter -> unit -> unit
+
+  val pp_close : F.formatter -> unit -> unit
+
+  val pp_rules : F.formatter -> elt -> unit
+
+  val pp : F.formatter -> elt -> unit
+end
+
+module MakeSarifListPrinter (P : sig
+  type elt
+
+  val rules_to_string : elt -> string option
+
+  val to_string : elt -> string option
+end) : PrinterSarif with type elt = P.elt = struct
+  include P
```

Comment:
I think it's enough to `open P` here, but if this is converted to a module then the functions in the argument of `SarifIssuePrinter` will go here instead of that anyway.